### PR TITLE
Fix multipart cache

### PR DESCRIPTION
**Pull request by pavelzubarev**
## Checklist
- [X] I have added/updated the necessary documentation for the changes.
- [ ] I have added/updated the tests for the changes.
- [X] I have updated the poetry version. Refer [version upgrades.](https://python-poetry.org/docs/cli/#version)
- [X] I have updated the version in `__init__.py`
- [X] I have updated the [CHANGELOG.md](https://github.com/cognitedata/cdffs/blob/main/CHANGELOG.md)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -75,7 +75,7 @@ twine = "^4.0.2"
 pydantic = "^2.4.2"
 python-dotenv = "^1.0.0"
 pydantic-settings = "^2.0.3"
-retry = "^0.9.2"
+tenacity = "^8.2.3"
```
Comment (vminfant) [inline pyproject.toml:78]: Good :)

File: cognite/cdffs/spec.py [hunk 2]
```diff
@@ -719,19 +746,36 @@ def strategy_submit(data: bytes, index: int) -> None:
...
-            total_size = self.write_strategy.merge_chunks()
 
-            self.cognite_client.files.update(
-                item=FileMetadataUpdate(external_id=self.external_id).metadata.add({"size": total_size})
+            @retry(
+                stop=stop_after_attempt(5),
+                wait=wait_fixed(0.5),
+                after=after_log(logging.getLogger("cdffs"), logging.INFO),
             )
+            def safe_merge() -> int:
```
Comment (vminfant) [inline cognite/cdffs/spec.py:755]: Do we really need this nested function here?

File: cognite/cdffs/spec.py [hunk 3]
```diff
@@ -719,19 +746,36 @@ def strategy_submit(data: bytes, index: int) -> None:
...
+                return self.write_strategy.merge_chunks()
+
+            total_size = safe_merge()
+
+            @retry(
+                stop=stop_after_attempt(5),
+                wait=wait_fixed(0.5),
+                after=after_log(logging.getLogger("cdffs"), logging.INFO),
+            )
+            def safe_file_update(_size: int) -> None:
```
Comment (vminfant) [inline cognite/cdffs/spec.py:765]: Same as above! Do we really need this nested function here?

**Review by vminfant — APPROVED** [review]
Great work! Some minor comments.

File: cognite/cdffs/spec.py (hunk 2 above)
Comment (pavelzubarev) [reply cognite/cdffs/spec.py:755]: In case something fails inside the strategy merge it actually can be retried

File: cognite/cdffs/spec.py (hunk 3 above)
Comment (pavelzubarev) [reply cognite/cdffs/spec.py:765]: Yes. to control the scope of retry

File: cognite/cdffs/spec.py (hunk 2 above)
Comment (pavelzubarev) [reply cognite/cdffs/spec.py:755]: tenacity comes with attributes only, so having smaller retry scope is better, I think

**Pull request merged**